rust-runtime/aws-smithy-runtime/src/client/connectors/connection_poisoning.rs

```diff
@@ -72,14 +73,10 @@ impl Interceptor for ConnectionPoisoningInterceptor {
             .map(RetryConfig::reconnect_mode)
             .unwrap_or(ReconnectMode::ReconnectOnTransientError);
         let captured_connection = cfg.load::<CaptureSmithyConnectionWrapper>().cloned();
-        let retry_classifiers = runtime_components
-            .retry_classifiers()
-            .ok_or("retry classifiers are required for connection poisoning to work")?;
-
-        let error_is_transient = retry_classifiers
-            .classify_retry(context.inner_mut())
-            .map(|reason| reason == RetryReason::Error(ErrorKind::TransientError))
-            .unwrap_or_default();
+        let retry_classifier_result =
+            run_classifiers_on_ctx(runtime_components.retry_classifiers(), context.inner_mut());
```

Comment:
seems odd to be passing a mutable reference to context here

Reply:
It's b/c I need this to be the actual context type, not `BeforeDeserializationInterceptorContextMut` and because that's a `mut` context type, it doesn't have an `inner` method, only an `inner_mut` method. Want me to add an `inner` method?

Reply:
never mind, John changed the interceptor context object used here so the point is moot.